File: client/app/src/main/java/ca/josephroque/hivemind/game/Player.kt
package ca.josephroque.hivemind.game

/**
 * Copyright (C) 2019 Joseph Roque
 *
 * The available players.
 */
enum class Player {
    White, Black;

    val next: Player
        get() {
            return when (this) {
                White -> Black
                Black -> White
            }
        }
}

File: engine/Sources/HiveMindEngine/State/Player.swift
//
//  Player.swift
//  HiveMind
//
//  Created by Joseph Roque on 2019-02-08.
//  Copyright © 2019 Joseph Roque. All rights reserved.
//

import Foundation

/// Players of the game
public enum Player {
	/// The first player, white
	case white
	/// The second player, black
	case black

	/// Returns the next player
	public var next: Player {
		switch self {
		case .black: return .white
		case .white: return .black
		}
	}
}
